flink-yarn/src/main/java/org/apache/flink/yarn/YarnApplicationFileUploader.java

```diff
@@ -388,13 +388,20 @@ private Path copyToRemoteApplicationDir(
                 (relativeDstPath.isEmpty() ? "" : relativeDstPath + "/") + localSrcPath.getName();
         final Path dst = new Path(applicationDir, suffix);
 
+        final Path localSrcPathWithScheme;
+        if (localSrcPath.toUri().getScheme() == null) {
```

Comment:
Not sure about the internal details of the Uri parsing but `isNullOrWhitespaceOnly` sounds more safe.

Reply:
Thank you, I agree, I've changed it